sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/TokenManager.java

```diff
@@ -88,4 +88,17 @@ public synchronized void removeCachedToken(String partitionName) {
         SharedPreferencesManager.putStringSet(Constants.PARTITION_NAMES, partitionNamesSet);
         SharedPreferencesManager.remove(partitionName);
     }
+
+    /**
+     * Remove all the cached access tokens for all partition names.
+     */
+    @SuppressWarnings("WeakerAccess")
+    public synchronized void removeAllCachedTokens() {
+        Set<String> partitionNamesSet = getPartitionNames();

```

Comment:
If user signs out, we must keep tokens related to read-only partitions

Reply:
@guperrot how can we determine from the SDK side whetehr the partition name is read-only?

Reply:
I think it's just comparing the name to Constants.READONLY, is that correct @YiMysty ?

Reply:
yup!